### PR TITLE
test: fix `ESLint` tests

### PR DESCRIPTION
**Pull request by fasttime**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

Fix unit tests

#### What changes did you make? (Give an overview)

This PR fixes several unit tests in tests/lib/eslint/eslint.js.

### CODE REVIEW

File: tests/lib/eslint/eslint.js [hunk 1]
```diff
@@ -928,6 +928,7 @@ describe("ESLint", () => {
 
             it("should throw if eslint.config.js file is not present even if overrideConfig was passed", async () => {
                 eslint = new ESLint({
+                    flags,
```
Comment (fasttime) [inline tests/lib/eslint/eslint.js:931]: It looks like this option was accidentally omitted here and in other unit tests in #18742.

File: tests/lib/eslint/eslint.js [hunk 2]
```diff
@@ -1525,26 +1526,28 @@ describe("ESLint", () => {
...
                     eslint = new ESLint({
+                        flags,
                         cwd: getFixturePath(".."),
                         overrideConfig: {
                             rules: {
                                 "no-unused-vars": 2
                             }
                         }
                     });
-                    await assert.rejects(() => eslint.lintFiles("fixtures/undef*.js"), /Could not find config file/u);
```
Comment (fasttime) [inline tests/lib/eslint/eslint.js:1535]: There's a config file in "fixtures", so this test wouldn't work with the `unstable_config_lookup_from_file` flag.

File: tests/lib/eslint/eslint.js [hunk 3]
```diff
@@ -5955,20 +5958,6 @@ describe("ESLint", () => {
...
-                        overrideConfigFile: true,
-                        fix: false,
-                        fixTypes: ["layout"]
-                    });
-                    const inputPath = getFixturePath("fix-types/fix-only-semi.js");
-                    const results = await eslint.lintFiles([inputPath]);
-
-                    assert.strictEqual(results[0].output, void 0);
-                });
-
```
Comment (fasttime) [inline tests/lib/eslint/eslint.js:5971]: Exact duplicate of the previous test.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**